### PR TITLE
Add test-utils with transfer wrapper

### PR DESCRIPTION
**Pull request by vlopes11**

### CODE REVIEW

File: contracts/test-utils/transfer-wrapper/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+[dependencies]
+canonical = "0.6"
+canonical_derive = "0.6"
+dusk-abi = "0.9.0-rc"
+rusk-abi = { path = "../../../rusk-abi" }
+dusk-bytes = "0.1"
+dusk-bls12_381 = { version = "0.8", default-features = false, features = ["canon"] }
+dusk-jubjub = { version = "0.10", default-features = false, features = ["canon"] }
+dusk-poseidon = { version = "0.21.0-rc", default-features = false, features = ["canon"] }
```
Comment (ZER0) [inline contracts/test-utils/transfer-wrapper/Cargo.toml:15]: We're landing a bunch of rc's, so at least for the coming PR that are still not merged, we should point to the proper version, without rc.

File: contracts/test-utils/transfer-wrapper/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    Ok((network, ssk))
+}
+
+pub fn transfer_state(
+    network: &NetworkState,
+) -> Result<TransferContract, VMError> {
+    network.get_contract_cast_state(&transfer_contract())
+}
+
+fn circuit_key(circuit_id: &[u8; 32]) -> Result<ProverKey, VMError> {
```
Comment (ZER0) [inline contracts/test-utils/transfer-wrapper/src/lib.rs:82]: What this functions does?

File: contracts/test-utils/transfer-wrapper/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    let pk = rusk_profile::keys_for(circuit_id)
+        .and_then(|keys| keys.get_prover())
+        .map_err(|e| VMError::ContractPanic(e.to_string()))?;
+    let pk = ProverKey::from_slice(pk.as_slice())
+        .map_err(|e| VMError::ContractPanic(e.to_string()))?;
+
+    Ok(pk)
+}
+
+fn prepare_execute<'a, R, I>(
```
Comment (ZER0) [inline contracts/test-utils/transfer-wrapper/src/lib.rs:92]: Add documentation: What this function does?

**Review by ZER0 — CHANGES_REQUESTED** [review]
In addition to the comments below, I notice a large misuse of the `VMError::ContractPanic`. This Error is returned only when a panic happen in WASM, so I'm not even sure why is used here.

**Review by CPerezz — CHANGES_REQUESTED** [review]

File: contracts/transfer/Cargo.toml [hunk 4]
```diff
@@ -27,7 +27,7 @@ rand = "0.8"
 transfer-circuits = { path = "../../circuits/transfer", features = ["builder"] }
 rusk-profile = { path = "../../rusk-profile" }
 dusk-plonk = "0.8"
-rusk-vm = "0.6.0-rc"
+rusk-vm = "0.6"
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply contracts/transfer/Cargo.toml:30]: This is not really part of the commit but a consequence of the merge. Better not touch stuff outside the scope of the test-utils

File: rusk/Cargo.toml [hunk 5]
```diff
@@ -41,7 +41,7 @@ rustc_tools_util = "0.2"
 rand = "0.8"
 lazy_static = "1.4"
 rusk-profile = { path = "../rusk-profile" }
-rusk-vm = "0.6.0-rc.0"
+rusk-vm = "0.6"
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply rusk/Cargo.toml:44]: Ditto; not part of the scope

File: test-utils/transfer-wrapper/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+[package]
+name = "transfer-wrapper"
+version = "0.1.0"
+authors = ["Victor Lopez <victor@dusk.network>"]
+edition = "2018"
+
+[dependencies]
+canonical = "0.6"
+canonical_derive = "0.6"
+dusk-abi = "0.9.0-rc"
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply test-utils/transfer-wrapper/Cargo.toml]: Fixed

File: contracts/test-utils/transfer-wrapper/src/lib.rs (hunk 3 above)
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/src/lib.rs]: This is private scope, but I added a comment

File: contracts/test-utils/transfer-wrapper/src/lib.rs (hunk 2 above)
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/src/lib.rs]: This is private scope, but I added a comment

File: contracts/test-utils/transfer-wrapper/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+    let proof = execute_proof
+        .gen_proof(&*PP, &pk, b"dusk-network")
+        .map_err(|e| VMError::ContractPanic(e.to_string()))?
+        .to_bytes()
+        .to_vec();
+
+    Ok((meter, anchor, nullifiers, fee, crossover, outputs, proof))
+}
+
+pub fn execute<'a, R, I>(
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/src/lib.rs]: Added

File: contracts/test-utils/transfer-wrapper/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+    let transfer = network
+        .deploy(transfer)
+        .map_err(|_| TransferError::ContractNotFound)?;
+
+    assert_eq!(transfer, transfer_contract());
+
+    Ok((network, ssk))
+}
+
+pub fn transfer_state(
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/src/lib.rs]: Added

File: contracts/test-utils/transfer-wrapper/src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+
+pub fn transfer_contract() -> ContractId {
+    ContractId::from([
+        0xd3, 0xf8, 0x7f, 0xfc, 0x1b, 0xc7, 0x43, 0x1d, 0xde, 0x81, 0x5f, 0xb1,
+        0xe1, 0x1b, 0xd0, 0xfe, 0x88, 0x37, 0x1a, 0x15, 0x4a, 0xec, 0x27, 0x5d,
+        0xed, 0x2, 0x4d, 0x8c, 0xc0, 0xf7, 0x99, 0x5f,
+    ])
+}
+
+pub fn genesis<R>(
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/src/lib.rs]: Added

File: contracts/test-utils/transfer-wrapper/src/lib.rs [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+    };
+}
+
+pub fn transfer_contract() -> ContractId {
+    ContractId::from([
+        0xd3, 0xf8, 0x7f, 0xfc, 0x1b, 0xc7, 0x43, 0x1d, 0xde, 0x81, 0x5f, 0xb1,
+        0xe1, 0x1b, 0xd0, 0xfe, 0x88, 0x37, 0x1a, 0x15, 0x4a, 0xec, 0x27, 0x5d,
+        0xed, 0x2, 0x4d, 0x8c, 0xc0, 0xf7, 0x99, 0x5f,
+    ])
+}
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/src/lib.rs]: Fixed

File: Cargo.toml [hunk 11]
```diff
@@ -5,6 +5,7 @@ members = [
     "circuits/transfer",
     "contracts/bid",
     "contracts/transfer",
+    "contracts/test-utils/transfer-wrapper",
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply Cargo.toml]: Fixed

File: contracts/test-utils/transfer-wrapper/Cargo.toml (hunk 1 above)
Comment (vlopes11) [reply contracts/test-utils/transfer-wrapper/Cargo.toml]: This crate contains the latest r3elease for all deps

**Review by CPerezz — APPROVED** [review]
It wouldn't be bad to add a bit of docs or a README at least.. but for the code, LGTM

**Review by CPerezz — APPROVED** [review]
LGTM

**Pull request merged**